### DISCUSSION TITLE
VoNR Guide with open5gs, srsRAN , and Kamilio

### DISCUSSION THREAD

**Discussion by AzeezEbrahim** [herlesupreeth/docker_open5gs#314] [General]
hey, 4 days ago kamilio published a video showing their VoNR support, so i wonder if anyone can help to configure it as well.
https://www.kamailio.org/w/2024/04/kamailio-powering-first-live-demo-of-5g-vonr-and-smsonr-with-oss-only/

i followed the steps with sa-vonr-deploy, and removed the checkout commit from the kamilio, but still can't make a call. 
what else? what is missing? or is the setup way different than what presented here?

**Comment by herlesupreeth** [comment]
Hey @AzeezEbrahim the configuration files used in that demo is not made public yet. In this repo, using sa-vonr-deploy.yaml you can successfully perform VoNR registration but calling is not possible since there is no N5 interface  to request QoS required for the call.

**Comment by AzeezEbrahim** [comment]
hey @herlesupreeth 
Thank you for your clarification.

Quick question, do they mention when are they going to release the docs for the phone calls? are they even going to in the near future? if so, are you going to write a step by step document about it?

**Comment by herlesupreeth** [comment]
> Quick question, do they mention when are they going to release the docs for the phone calls?

They plan to release the configuration files public once they are done with some level of testing. Not sure when. I am expecting in a month or two max.

> are you going to write a step by step document about it?

Definitely yes

**Comment by sathyag85** [comment]
I am also setting up VONR until registration is successful with the help of SA-vonr-deploy.yaml file. The call is not working because there is no N5 interface to request the QoS required for the call. Please update if the configuration files are available for the above.Thanks.

**Comment by JoseVOliveria** [comment]
Hello,
I was able to establish a call using the sa-vonr-deploy.yaml file. I did not add any configurations. I used two xiaomi phones with srsran in a USRPB210. I leave the pcap fila attached.

[call.pcap.zip](https://github.com/user-attachments/files/16254345/call.pcap.zip)

**Reply by sathyag85** [reply]
@JoseVOliveria  Is this a VONR call? Could you please share the trace along with the NGAP protocol?

**Reply by JoseVOliveria** [reply]
Hello, yes it is
Sure.
[Archive.zip](https://github.com/user-attachments/files/16263670/Archive.zip)

**Reply by sathyag85** [reply]
In your trace, there is no bearer allocation of 5QI = 1 happening during the call..

**Reply by JoseVOliveria** [reply]
There is no bearer allocation because there is no communication with the PCF for a Session Modification Request. Despite that, using the APN=ims and 5QI=5 the smartphone successfully registers in the IMS network and the call still goes through. To execute the registration we used an android app called Sipnetic.

**Reply by anhtrung87vn** [reply]
Hi @JoseVOliveria, which Xiaomi phone model and PLMN are you using for VoNR?

**Reply by JoseVOliveria** [reply]
Hi I was using the Redmi Model:M2007J22G and the PLMN 00101

**Comment by herlesupreeth** [comment]
For those who are interested in trying out VoNR calling- here is a branch https://github.com/herlesupreeth/docker_open5gs/tree/exp_5g_ims_pyhss which contains the necessary N5 interface implementatio for P-CSCF so that it can talk with PCF. Huge shoutout to @NUCLEAR-WAR for this efforts. Please try it out and let me know if it works

**Reply by BurmazDaniel** [reply]
I couldn't get this branch to run properly.
Can we maybe have a guide for 5g sa core+vonr?

**Reply by jmrepollo** [reply]
Hi @herlesupreeth,

I'm trying to setup using exp_5g_ims_pyhss branch, PCSCF is unable to communicate with SCP based on the logs of PCSCF. 

![image](https://github.com/user-attachments/assets/f58dfde7-8d21-4809-a3cb-3312b8f39f8f)

**Reply by herlesupreeth** [reply]
> I couldn't get this branch to run properly.

@BurmazDaniel what do you mean by this? what is the error you are facing?

@jmrepollo Did you change the IP of the SCP in .env file or P-CSCF configuration file?

**Reply by jmrepollo** [reply]
@herlesupreeth yes I changed it.

![image](https://github.com/user-attachments/assets/f89cc2c2-96fe-4199-a08e-5bc2e14f1023)

![image](https://github.com/user-attachments/assets/ff83073a-5482-40a8-bfe7-cf02dd7a88da)

**Reply by herlesupreeth** [reply]
@jmrepollo If you are not using docker_open5gs for deploying SCP then unfortunately I cannot be of much help. I would recommend using docker_open5gs as whole while experimenting with VoNR

**Reply by jmrepollo** [reply]
@herlesupreeth 

yes, I'm not using your setup for deploying Open5GS. I've installed it in a single container. I'll give your setup a try and provide feedback afterwards. Thanks!

**Reply by NUCLEAR-WAR** [reply]
I think there is one thing missing from the normal guide, in the guide it is written that to deploy use :

5G Core Network
docker compose -f sa-deploy.yaml up

this will run only 5G Core without the IMS, for the new changes you need to run :

5G Core Network + IMS 
docker compose -f sa-vonr-deploy.yaml up

this should run the 5GC and the IMS.

**Reply by BurmazDaniel** [reply]
@herlesupreeth
@NUCLEAR-WAR 

I managed to get it working properly. YEY. I had some docker interface issues. SMS is working fine, have not tested the voice yet but i have a question about ims_subscriber in pyhss oami.
What exactly is scscf_peer, scscf and scscf_realm in Create new IMS SUBSCRIBER? Should i replace with someting else?
I connot ping it from host machine. Shoud i try within the docker interface? Sorry for my stupid questions but i am trying to understand.
Thanks!

**Reply by NUCLEAR-WAR** [reply]
@BurmazDaniel  I would suggest to use the example from @herlesupreeth  when creating IMS Subscriber : 

{
    "imsi": "001010123456790",
    "msisdn": "9076543210",
    "sh_profile": "string",
    "scscf_peer": "scscf.ims.mnc001.mcc001.3gppnetwork.org",
    "msisdn_list": "[9076543210]",
    "ifc_path": "default_ifc.xml",
    "scscf": "sip:scscf.ims.mnc001.mcc001.3gppnetwork.org:6060",
    "scscf_realm": "ims.mnc001.mcc001.3gppnetwork.org"
}
dont forget to change the IMSI/MSISDN to suite your SIM Card info and your setup.
also you need to make sure that the DNS Server you use knows FQDNs, otherwise it will not work, if you use the Docker solution this should be no problem, but you should deploy with sa-vonr-deploy.yaml  so the IMS also starts.

**Reply by BurmazDaniel** [reply]
@NUCLEAR-WAR voice is working ok ish. I have some issues. 
```
icscf      |  2(35) ERROR: <script>: $ru => sip:8821100002@ims.mnc001.mcc001.3gppnetwork.org;user=phone
icscf      |  2(35) INFO: ims_icscf [cxdx_lir.c:70]: create_lia_return_code(): created AVP successfully : [lia_return_code]
```
after some tests answering the phone calls
also the smf seems to lose the qos 
```
smf        | 09/02 12:17:38.023: [smf] ERROR: [imsi-001010000067642:1] No QoS flow (../src/smf/ngap-handler.c:350)
smf        | 09/02 12:17:38.024: [smf] ERROR: [imsi-001010000067642:1] Cannot handle NGAP message (../src/smf/gsm-sm.c:1237)
smf        | 09/02 12:17:38.024: [smf] ERROR: [imsi-001010000067642:1] State machine exception (../src/smf/gsm-sm.c:2168)
```
and after that the pcscf seems to lose the connection too
```
pcscf      | 85(128) ERROR: <script>:   request sent to sip:4239b1e6-0e76-49f9-a99f-591489694395@10.46.0.9:43074: Fail Counter is 1
```
After a complete reboot of docker containers the flow goes back to normal. I dont know what is causing this. I will paste a more detailed log asap.

**Reply by NUCLEAR-WAR** [reply]
I think this issue is related to the issue [768](https://github.com/srsran/srsRAN_Project/issues/768) the current srsGNB does not support modifying QoS Session that leads to detach the UE and then a re-attach sometimes gNodeB just stops working and need to be restarted, I will provide today the logs to the srsRAN.
Try to only restart the gNB instead of the whole docker.

**Reply by BurmazDaniel** [reply]
@NUCLEAR-WAR dose oai overcome this issue somehow?

UEs connect but when i try to make a call the gnb crashes.
```
[NR_RRC]   Cant' handle NGAP_ProtocolIE_ID_id_PDUSessionAggregateMaximumBitRate
[NR_RRC]   could not found protocolIEs id 139
[NR_RRC]   index 0, QOS flow 0, 5QI 5 
[NR_RRC]   add NAS info with size 85 (pdusession id 1)
[NR_RRC]   [gNB 0] Frame 0, Logical Channel DL-DCCH, Generate RRCReconfiguration (bytes 99, UE RNTI a397)
[NR_RRC]   UE 1: Receive RRC Reconfiguration Complete message (xid 0)
[NR_RRC]   send message NGAP_PDUSESSION_MODIFY_RESP 
[NR_RRC]   update pdu session 1 
[NR_RRC]   Modify Resp (msg index 0, pdu session index 0, status 2, xid 0): nb_of_pduSessions 1,  pdusession_id 1
```

**Reply by NUCLEAR-WAR** [reply]
Try to compile the last version of gNB from srsRAN Repo, it will be less crushing, or use the master branch's gNB it also has the last version.

I just updated the issue on srsRAN Repo with files requested, I hope they can add the support for Modify QoS Sessions

**Reply by NUCLEAR-WAR** [reply]
and regarding OAI, it also does not support QoS modification, there Repo also not easelly accessible to report the issue

**Reply by BurmazDaniel** [reply]
@NUCLEAR-WAR thank you. Yes im using the lates commit [4ac5300](https://github.com/srsran/srsRAN_Project/commit/4ac5300d4927b5199af69e6bc2e55d061fc33652) for srsRAN, compiled from source. Ive tested quite a lot the vonr and it seems to be more stable using mimo 2x2. The packet ok/nok rate seem to be the same but the vonr dropouts were significantly reduced. Quite an interesting behavior.

**Reply by NUCLEAR-WAR** [reply]
this is because srsRAN does not know what to do with the modify QoS Request, what is happening is that the device will detach and reattach again, you will see it in the trace als, the last version can hold longer but it has at the end the same fate, sometimes it silently stops working after few requests and need to be restarted .

I provided more logs to the issue in srsRAN, the whole N5 with Kamailio is new, and not been tried before so I was expecting some issues with some elements.

may I ask if you tried a call between two UEs and how was the signalling ? can you provide me with a Trace ?

**Reply by fatima6299** [reply]
Dans le cadre de mon memoire je dois mettre en place IMS dans le reseau 5G SA, pouvez vous m'aidez

**Reply by Silen008** [reply]
https://github.com/herlesupreeth/docker_open5gs/tree/exp_5g_ims_pyhss why this branch is 404?

**Reply by herlesupreeth** [reply]
@Silen008 because changes in that branch is now merged to master branch. And if you use master branch you should have VoNR calling feature

**Comment by AzeezEbrahim** [comment]
Hi @herlesupreeth, i am still running into issues with Vonr calling. do you have a complete guide/instructions on how to setup the vonr call.

**Reply by herlesupreeth** [reply]
Hi @AzeezEbrahim , I just pushed a commit to exp_5g_ims_pyhss branch which has the fix for VoNR calling issue (needed to use latest srsRAN_Project code)

All you need to do is switch to exp_5g_ims_pyhss, compile all the base images and deploy using sa-vonr-deploy.yaml thats all. If you still face issue please send a pcap of scenario where UE attaches to the network + attempts a call

**Reply by AzeezEbrahim** [reply]
Hi @herlesupreeth thanks for the quick response !

i did use exp_5g_ims_pyhss branch, and i did all the steps, sms is working fine and i see voLTE sign. but when i try to call this error appear in the pcscf.

![image](https://github.com/user-attachments/assets/03b2c8a5-dcbe-4f53-b54c-462713111c02)

and there is also this error in the pcf
![image](https://github.com/user-attachments/assets/b58d35b4-543f-4642-9be4-14f964686be1)`
Questions:
- Should i turn off voLTE option in the phones and only turn on VoNR?
- Should i set the 
apn to internet in the phones without the ims? 
- Regarding the pcaps what are the filters should i use?

[vonr.zip](https://github.com/user-attachments/files/18884344/vonr.zip)

**Reply by NUCLEAR-WAR** [reply]
Hi @AzeezEbrahim, did you configured The APNs and the QCI in WebGui? the error from PCF suggest that you did not.

configure the MSISDNs like this Picture using the WebGui under http://<DOCKER_HOST_IP>:9999).

![subscriber-type-ipv4](https://github.com/user-attachments/assets/6ea9770b-5c5d-499a-a5c6-e484563d8034)

**Reply by AzeezEbrahim** [reply]
Thanks you saved my day @NUCLEAR-WAR.

it seems that my issue is related to last row in the PCC Rules, it didn't work when it was configured as "unlimited". but once i change it to 128 Kbps, do you know the reason for that? 
![image](https://github.com/user-attachments/assets/5a6b3c89-7db2-4bd8-aba3-583ce816b1bc)
changes:
![image](https://github.com/user-attachments/assets/4629cadf-f203-46fe-8806-b8d0f4a42ccc)

Thanks again !

**Reply by herlesupreeth** [reply]
Thats because GBR and MBR are needed for setting up dedicated bearers. I am guessing setting to unlimited (not set value in webui) probably wont create those bearers

**Comment by sayandeeps** [comment]
is there any detailed guide for vonr with ogs as well as i do not have any physical devices can i use ueransim or something similar i need help for this please help

**Reply by herlesupreeth** [reply]
There is no guide yet. But if you follow the Dockerfile in ims_base folder you should be able to use the latest kamailio image and also use the cfg files in this repo to get VoNR working. However, I dont think its possible to use UERANSIM or similar to test VoNR since it would require tight integration with IMS to request a QoS etc.

**Reply by prajjwal02** [reply]
Hi @herlesupreeth 
What would be the best way to test VoNR registration & Call Setup if one doesnt have SDR and real UE. 
I am using the same repo and have deployed open5gs+kamalio using 5docker compose -f sa-vonr-deploy.yaml up file. But how should i test VoNR from here on ?
Can you please assist me here
Thanks

**Comment by evershalik** [comment]
Hi @herlesupreeth , I was trying to deploy kamailio with open5gs(VoNR deployment), following your repo. I have read all the chats above and fixed some little issues you suggested like the QoS configuration in dashboard. For now I have some question in my mind:
1. in my case the UE shows VoLTE and another UE shows HD(but i am using `sa-vonr-deploy.yaml` file), here i want to ask, do we need to make any changes on the UE side to make the vonr work properly? Also there is a option called `enable 4g calling` but when i disable this expecting vonr to work, then after this VoLTE/HD also disappear(in both UEs). Is there any need to make specific changes on the UE side(something like any application, dialer code or root).
<img width="269" alt="image" src="https://github.com/user-attachments/assets/6128e3e0-7e7e-441b-b75e-d845705251e0" />
<img width="555" alt="image" src="https://github.com/user-attachments/assets/ebbe670d-e424-4a82-8550-ba7c02ded918" />

2. I think I am able to send SMS from one UE but unable to receive on another UE according to this screenshot:
<img width="660" alt="image" src="https://github.com/user-attachments/assets/c6ebeae7-3a9d-45f7-b0ce-e9045a5aa525" />

Also i am unable to make any call , I tried to make call attempt but it just dropped, i am attaching the pacps and some logs for the same. 

[sms_and_call_trying_kamailio_open5gs_13_april_2025.pcapng.zip](https://github.com/user-attachments/files/19725029/sms_and_call_trying_kamailio_open5gs_13_april_2025.pcapng.zip)
[logs.zip](https://github.com/user-attachments/files/19725035/logs.zip)

And these are my subscriber details:
<img width="844" alt="image" src="https://github.com/user-attachments/assets/887d5b2f-fe81-4edb-bc24-9a82a08e3f6d" />

I would really appreciate you help. Thank you

**Reply by herlesupreeth** [reply]
> in my case the UE shows VoLTE and another UE shows HD(but i am using sa-vonr-deploy.yaml file), here i want to ask, do we need to make any changes on the UE side to make the vonr work properly? 

its normal that UE shows VoLTE rather than VoNR since it shows the same for me on some phones. I dont what's the prerequisite for UE to show VoNR. But in principle there is no difference in working principle between VoNR/VoLTE from the UE perspective (only changes is on network side)

> Also there is a option called enable 4g calling but when i disable this expecting vonr to work, then after this VoLTE/HD also disappear(in both UEs)

Its correct behavior. Disabling "enable 4G calling" disables IMS and expects phone to have 2G connectivity for placing the call and since its not there as part of docker_open5gs deployment the UE shuts off

> Also i am unable to make any call , I tried to make call attempt but it just dropped, i am attaching the pacps and some logs for the same.

Are you saying that you are not able to make the call even when UE shows VoLTE/HD ?

**Reply by evershalik** [reply]
Yes, even after VOLTE/HD symbol on my mobile device, I am unable to make any phone calls when I try to it drops the connection. Also as you can see the above screenshot, I can send SMS from UE 1 but unable to receive the SMS on the another UE 2 so I'm not sure what is the issue there? I have already attached the pcaps and logs. Can you help me out what is the issue.

**Reply by herlesupreeth** [reply]
Can you please take a clean pcap of following scenario:

1. Bring down all services
2. Start wireshark/tcpdump
3. Bring up all services
4. Start gNB
5. Attach two UEs one after another
6. Attempt a call from one to another
7. Stop wireshark/tcpdump

Send me the captured pcap

Btw, are you using srsRAN_Project gNB? If so can you tell me whether you are using the deployment file in this repo or are you running gNB on the host? Also, command used to run gNB

**Reply by evershalik** [reply]
Hi @herlesupreeth,
Here are the PCAPs for the scenario you suggested. I followed the same steps you mentioned to capture the PCAPs and tried to call from UE1 to UE2 and vice versa. 

**PCAP**: [open5gs_kamailio_voice_call_test_19_april_2025.pcap.zip](https://github.com/user-attachments/files/19821170/open5gs_kamailio_voice_call_test_19_april_2025.pcap.zip)

For you reference:

**UE1** has MSISDN: `00000002`
**UE2** has MSISDN: `00000003`

On **UE1**, `VoLTE` was apprearing, but
On **UE2**, `HD` was appearing(sometime appear sometime not)

And regarding the **srsRAN**, I am using the same deployment file in this repo(`srsgnb.yaml`) and command used to run the gnb is `docker compose -f srsgnb.yaml up -d && docker container attach srsgnb`

**Reply by evershalik** [reply]
@herlesupreeth Can you help me here?

**Reply by herlesupreeth** [reply]
I am out of town. I will take a look once I am back

**Reply by herlesupreeth** [reply]
@evershalik From your pcap attached I see that you have manually created "ims" DNN/APN on the phone and selected it. You should rather create a DNN/APN called "internet" and select it. This is the correct way. 

Additionally, I see that your second UE in the pcap does have IMS enabled and is attaching to 5G network using "ims" DNN/APN (like I mentioned above). As a result of this your calls are failing. (NOTICE that there is no SIP REGISTER for second UE).

Only your NOTHING phone is successfully connected to IMS.

![image](https://github.com/user-attachments/assets/35f96762-f16a-462d-be45-c12b611df9f4)

**Reply by evershalik** [reply]
You are right, I manually created the `ims` DNN/APN on both the phones. However I also tried using the `internet` DNN/APN and attempted to call from UE 1 to UE 2, still the call wasn’t successful. 
Can you clear what is meant by the statement “your second UE in the `pcap` does have IMS enabled”, I mean do I need to do something different on second UE, I did the same procedure as I did with UE 1. And yes you are right maybe there is no SIP REGISTER for second UE because of the ims DNN/APN. Do you see any other issues in the `pcap`? I am asking this because I tried calling with `internet` DNN/APN in both the UEs, but still the call was dropping.

**Reply by herlesupreeth** [reply]
> Can you clear what is meant by the statement “your second UE in the pcap does have IMS enabled”, I mean do I need to do something different on second UE, I did the same procedure as I did with UE 1.

My bad, I intended to write "Additionally, I see that your second UE in the pcap does NOT have IMS enabled..." If its an Android phone then you could try running CoIMS as mentioned in this wiki - https://github.com/herlesupreeth/CoIMS_Wiki (its not a 100% working solution but definitely worth a try)

**Comment by qmodem-tech** [comment]
@herlesupreeth, hi! I deployed and started your project with sa-vonr-deploy.yaml. I have SIP registration on both UEs. And I made just only one successful call with 30 sec duration because of RTP/RTCP timeout :
`Proxy-Require: sec-agree 
 Reason: RELEASE_CAUSE ;cause=2 ;text="RTP/RTCP time-out" 
 User-Agent: asus_ASUS_I007D_18_1055_2111_171 
 Content-Length: 0 
 P-Access-Network-Info: 3GPP-NR-TDD; utran-cell-id-3gpp=2089300000100066C000`
 
Attempting to redial doesn't yield any results. After restarting all docker containers  - its working just for 1 call with a 30 sec timeout for RTP/RTCP.
Any ideas how to solve it?

**Reply by herlesupreeth** [reply]
Hmm, my guess is that QoS is not met for the VoNR call (maybe setup of the QoS flow for audio call is failing)

> its working just for 1 call with a 30 sec timeout for RTP/RTCP.

Cant say for sure but if you could upload the pcap taken on docker_open5gs machine when you attempt the call until the call gets disconnect it will help debug this

**Reply by qmodem-tech** [reply]
Hi. Thank you for your prompt response. Here is the attached file with the successful call, but with limited duration due to the RTP/RTCP timeout error. And one more thing: another call is not working; I need to stop open5Gs and srsgnb and rerun it for the second call.
[rtp_rtcp_timeout.zip](https://github.com/user-attachments/files/20772799/rtp_rtcp_timeout.zip)

**Reply by qmodem-tech** [reply]
@herlesupreeth ,seems you were right . Now it's more stable after changing location of the UEs and increasing tx gain in usrp.

**Reply by herlesupreeth** [reply]
![image](https://github.com/user-attachments/assets/883af653-8e88-48da-b0a1-d2f276a84949)

Your gNB is rejecting PDU session modification (while adding QoS flows for Audio call). May I know which gNB you are using? If you are using srsRAN_Project gNB then you would need to use the qos.yml file as well in order for QoS flow with 5QI 1 to be setup

**Reply by qmodem-tech** [reply]
@herlesupreeth, I'm using srsRAN on the same host as Open5GS. I had a problem with the antenna connection, as I mentioned. In NSG in the Events tab, there were a lots of RACH failures and RRC reestablishments. Then I resolved this issue, and it now appears to be more stable. I did several manual and automated calls, and it seems OK. Thank you!

**Comment by Benie140902** [comment]
Is it mandatory to push user data in pyHSS for VONR if so i am unable to push auc for more than one user

**Reply by herlesupreeth** [reply]
yes, its mandatory to push Subscriber data to pyHSS. I have answered regarding AuC in your other disucssion

**Comment by qmodem-tech** [comment]
Hi. After succeeding on the first server, Im trying to deploy on another one and seeing the following:
```
WARNING: Failed to connect to server [mongodb://172.22.0.2/open5gs] (../lib/dbi/ogs-mongoc.c:122)
pcf      | 07/29 12:25:09.070: [app] ERROR: Failed to initialize PCF (../src/pcf/app.c:28)
udr      | 07/29 12:25:09.102: [dbi] WARNING: Failed to connect to server [mongodb://172.22.0.2/open5gs] (../lib/dbi/ogs-mongoc.c:122)
udr      | 07/29 12:25:09.102: [app] WARNING: Failed to initialize UDR (../src/udr/app.c:28)
pcf exited with code 0
udr exited with code 0
webui    | MongooseServerSelectionError: Server selection timed out after 1000 ms
webui    |     at NativeConnection.Connection.openUri (/open5gs/webui/node_modules/mongoose/lib/connection.js:847:32)
webui    |     at /open5gs/webui/node_modules/mongoose/lib/index.js:351:10
webui    |     at /open5gs/webui/node_modules/mongoose/lib/helpers/promiseOrCallback.js:32:5
webui    |     at new Promise (<anonymous>)
webui    |     at promiseOrCallback (/open5gs/webui/node_modules/mongoose/lib/helpers/promiseOrCallback.js:31:10)
webui    |     at Mongoose._promiseOrCallback (/open5gs/webui/node_modules/mongoose/lib/index.js:1149:10)
webui    |     at Mongoose.connect (/open5gs/webui/node_modules/mongoose/lib/index.js:350:20)
webui    |     at /open5gs/webui/server/index.js:39:29
webui    |     at Generator.next (<anonymous>)
webui    |     at onFulfilled (/open5gs/webui/node_modules/co/index.js:65:19)
webui    |     at processTicksAndRejections (node:internal/process/task_queues:9
```
What's wrong? I am doing the same rollout actions.

**Reply by herlesupreeth** [reply]
check the logs on mongodb container

**Reply by qmodem-tech** [reply]
Hi. In starting logs found:  WARNING: MongoDB 5.0+ requires a CPU with AVX support, and your current system does not appear to have that!
mongo    |   see https://jira.mongodb.org/browse/SERVER-54407
mongo    |   see also https://www.mongodb.com/community/forums/t/mongodb-5-0-cpu-intel-g4650-compatibility/116610/2
mongo    |   see also https://github.com/docker-library/mongo/issues/485#issuecomment-891991814

Any ideas how to overcome it?

**Reply by qmodem-tech** [reply]
I did it. Changing cpu type in VM config

**Comment by qmodem-tech** [comment]
@herlesupreeth Now I have another question. I have real gNb and it tries to establish connection to the AMF, but on SCTP level I see that gNB sends INIT, but the host with the IP where the AMF is located is answering with ABORT.
<img width="237" height="199" alt="image" src="https://github.com/user-attachments/assets/7308f134-67a6-4cf7-a50b-719a116d836e" />
Any ideas? I followed the instruction with separate deployment. Thanks!

**Reply by herlesupreeth** [reply]
Did you publish (uncomment as mentioned in README) the port 38412 ? If not, do it and also uncomment UPF port in docker compose file as mentioned in README and set UPF_ADVERTISE_IP accordingly

**Comment by qmodem-tech** [comment]
Hi. I have following issue: two setup - open5gs+srsran everything perfect. Then Open5gs + real gNb in AMF I see following:

amf  | 09/10 10:23:44.802: [amf] INFO: InitialUEMessage (../src/amf/ngap-handler.c:437)
amf  | 09/10 10:23:44.802: [amf] INFO: [Added] Number of gNB-UEs is now 2 (../src/amf/context.c:2789)
amf  | 09/10 10:23:44.802: [amf] INFO:     RAN_UE_NGAP_ID[16790020] AMF_UE_NGAP_ID[182] TAC[7] CellID[0x9d7800b] (../src/amf/ngap-handler.c:598)
amf  | 09/10 10:23:44.802: [amf] INFO: [suci-0-208-93-0-0-0-0000000001] Unknown UE by SUCI (../src/amf/context.c:1906)
amf  | 09/10 10:23:44.802: [amf] INFO: [Added] Number of AMF-UEs is now 2 (../src/amf/context.c:1682)
amf  | 09/10 10:23:44.802: [gmm] INFO: Registration request (../src/amf/gmm-sm.c:1333)
amf  | 09/10 10:23:44.802: [gmm] INFO: [suci-0-208-93-0-0-0-0000000001]    SUCI (../src/amf/gmm-handler.c:174)
amf  | 09/10 10:23:44.804: [gmm] ERROR: [suci-0-208-93-0-0-0-0000000001] HTTP response error [400] (../src/amf/gmm-sm.c:1919)
amf  | 09/10 10:23:44.804: [amf] WARNING: [suci-0-208-93-0-0-0-0000000001] Registration reject [95] (../src/amf/nas-path.c:211)
amf  | 09/10 10:23:44.804: [gmm] WARNING: [(null)] Failure in transaction; no context restoration. (../src/amf/gmm-sm.c:1927)
amf  | 09/10 10:23:45.228: [amf] ERROR: Not implemented(choice:1, proc:19) (../src/amf/ngap-sm.c:128)
amf  | 09/10 10:23:45.230: [amf] INFO: UE Context Release [Action:3] (../src/amf/ngap-handler.c:1733)
amf  | 09/10 10:23:45.230: [amf] INFO:     RAN_UE_NGAP_ID[16790020] AMF_UE_NGAP_ID[182] (../src/amf/ngap-handler.c:1734)
amf  | 09/10 10:23:45.230: [amf] INFO:     SUCI[suci-0-208-93-0-0-0-0000000001] (../src/amf/ngap-handler.c:1738)
amf  | 09/10 10:23:45.230: [amf] INFO: [Removed] Number of gNB-UEs is now 1 (../src/amf/context.c:2796)
amf  | 09/10 10:23:45.230: [amf] INFO: [Removed] Number of AMF-UEs is now 1 (../src/amf/context.c:1780)

Same SIM card and same provisioning. What's wrong with this? 
Thanks

**Reply by herlesupreeth** [reply]
Can you please post a pcap taken on the machine running docker_open5gs when using real gNB (dont use any capture filters)?

**Reply by qmodem-tech** [reply]
Yes, here is attach.
[SAnoReg.zip](https://github.com/user-attachments/files/22273449/SAnoReg.zip)

**Reply by qmodem-tech** [reply]
Here how I see it from the real UE
 ![image](https://github.com/user-attachments/assets/7389b2f9-b705-40a0-9182-bb86ba29ae30)

**Reply by qmodem-tech** [reply]
@herlesupreeth , hi ! Any ideas ?

**Reply by herlesupreeth** [reply]
please try to keep this discussion clean. I will try to answer whenever I can so please be patient.

Hmm, I never saw a InitialUeMessage with RegistrationRequest (packet 214) with so few contents. Even more strange is your SIM PLMN is 208 93 but the gNB is having the following

```
UserLocationInformation: userLocationInformationNR (1)
    userLocationInformationNR
        nR-CGI
            pLMNIdentity: 64f010
                Mobile Country Code (MCC): China (460)
                Mobile Network Code (MNC): China Unicom (01)
            0000 0000 1001 1101 0111 1000 0000 0000 1011 .... = nRCellIdentity: 0x009d7800b
        tAI
            pLMNIdentity: 64f010
                Mobile Country Code (MCC): China (460)
                Mobile Network Code (MNC): China Unicom (01)
            tAC: 7 (0x000007)
        timeStamp: ec6d1ffd (Sep 11, 2025 10:16:29 UTC)

```

Can you please check your gNB configuration?

**Reply by qmodem-tech** [reply]
Yes, provisioned SIM card has 20893 and gnb has 46001. So its roaming case should work. Isn't it ?

**Comment by Funny-Potato** [comment]
@herlesupreeth I'm sorry for an irrelevant question. Is there any guide on how to setup a VoLTE call with Kamailio? I made [another discussion](https://github.com/herlesupreeth/docker_open5gs/discussions/533) about this a few days ago. The Open5gs tutorials give me the errors I've provided.

**Reply by herlesupreeth** [reply]
@Funny-Potato I haven't had time to look at the discussion topic you raise. Sorry about that. I will try to take a look today or max. by weekend.

**Comment by punith-2000** [comment]
I am trying to DO VoNR registration but PDUsessionResourceResponse not coming only gettin request again after sending accept also
[VoNR_log.zip](https://github.com/user-attachments/files/22539587/VoNR_log.zip)
check above pcap log
[VoNR_NAS_log.zip](https://github.com/user-attachments/files/22539641/VoNR_NAS_log.zip)

**Reply by herlesupreeth** [reply]
@punith-2000 there is no PDUSessionResourceResponse from the gNB so you may want to check the gNB logs/pcaps for errors.

**Comment by dominicpeter** [comment]
I am trying to get PDUSessionResourceRequest for Vonr registration,what may be the issue, Where  I need to modify,I enable the vonr in phone.I put the apn as internet also.
[capture.zip](https://github.com/user-attachments/files/23846533/capture.zip)

**Reply by herlesupreeth** [reply]
@dominicpeter Looks like IMS is not enabled on your UE. Are you by any chance using Samsung phone? If not, then you could try CoIMS method https://github.com/herlesupreeth/CoIMS_Wiki to force enable IMS on your phone

**Reply by dominicpeter** [reply]
I am using Samsung A14 5g phone.In that phone Vonr is not showing,only volte is showing.what change I want to make in UE.But I put real time sim it shows Vonr.what may the issue.

**Comment by dominicpeter** [comment]
Hi @herlesupreeth 

I am trying do emergency call using OAI + Open5gs+kamailio in VoNR in which SIP transaction complete but RTP data not coming in suspect that Modify Request not complete.
But in 2 mobile phone call it is working 
Note: I am using custom PSAP app for ecall
Here is the log for both
[pcap_log.zip](https://github.com/user-attachments/files/25071241/pcap_log.zip)

Please help me with this

**Reply by NUCLEAR-WAR** [reply]
If I remember correctly, OAI does not support to date modifying PDUSession, it only create it, a later modifying it due to added/removed streams will fail leading to UE re-attach, during the development of the N5 Logic I tried to submit a Ticket but there repo is very restricted, so I stuck srsRAN, they added the support for modifying the session at that time, srsRAN is in my opinion more better and open to the community than OAI.

**Reply by herlesupreeth** [reply]
@dominicpeter I dont see any issue in the NGAP signalling and neither the SIP signalling. Have you added static route from your custom PSAP to UPF so that RTP traffic from PSAP gets routed properly back to UE?
